src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPIJUtils.java

```diff
@@ -319,15 +319,10 @@ public static void applyEdits(Editor editor, Document document, List<TextEdit> e
         ApplicationManager.getApplication().runWriteAction(() -> edits.forEach(edit -> applyEdit(editor, edit, document)));
     }
 
-    public static boolean hasCapability(Either<Boolean, ? extends Object> eitherCapability) {
-        if(eitherCapability != null) {
-            if (eitherCapability.isLeft()) {
-                return eitherCapability.getLeft();
-            } else {
-                return eitherCapability.getRight() != null;
-            }
-        } else {
+    public static boolean hasCapability(final Either<Boolean, ? extends Object> eitherCapability) {
+        if(eitherCapability == null) {
             return false;
         }
+        return eitherCapability.isRight() || eitherCapability.getLeft();
```

Comment:
if eitherCapability is neither right nor left, then returning eitherCapability.getLeft() will NPE on unboxing